docs/LLS_Metallicities-Wotta/getionisation.py

```diff
@@ -0,0 +1,371 @@
+    xsi3chian=interp['SiIII'](result['pdfs'])
+    hist,edge=np.histogram(xsi3chian,bins=base)
+    hist=hist/(1.*len(xsi3chian))
+    return (hist,np.median(xsi3chian))
+
+
+
+"""
+Compute the X_CII along the chains
+
+"""
+
+def getions_xc2(result,base,interp):
+
+    xC2chian=interp['CII'](result['pdfs'])
+    hist,edge=np.histogram(xC2chian,bins=base)
+    hist=hist/(1.*len(xC2chian))
+
+    return (hist,np.median(xC2chian))
+
+"""
+Compute the X_CIII along the chains
+
+"""
+
+def getions_xc3(result,base,interp):
+
+    xC3chian=interp['CIII'](result['pdfs'])
+    hist,edge=np.histogram(xC3chian,bins=base)
+    hist=hist/(1.*len(xC3chian))
+
+    return (hist,np.median(xC3chian))
+
+
+"""
+Compute the X_CIV along the chains
+
+"""
+
+def getions_xc4(result,base,interp):
+
+    xC4chian=interp['CIV'](result['pdfs'])
+    hist,edge=np.histogram(xC4chian,bins=base)
+    hist=hist/(1.*len(xC4chian))
+
+    return (hist,np.median(xC4chian))
+
+
+"""
+Initialise the interpolator over the grid
+
+"""
+
+def init_interpolator(grid):
+
+    
+    #open the grid 
+    fil=open(grid)
+    modl=pickle.load(fil)
+    fil.close()
+    
+    #define the dimension of the problem
+    mod_axistag=modl[0]
+    mod_axisval=[]
+    ndim=len(mod_axistag)
+    nmodels=1
+    for tt in mod_axistag:
+        nmodels=nmodels*(modl[1][tt]).size
+        #append axis value in a list
+        mod_axisval.append(modl[1][tt])
+   
+    #extract x_HI
+    xhigrid=modl[2]['HI']
+    #set up interpolator for this ions 
+    xhi_interp=interpolate.RegularGridInterpolator(mod_axisval,xhigrid,method='linear',bounds_error=False,fill_value=-np.inf)
+        
+    #extract x_SiII
+    xsi2grid=modl[2]['SiII']
+    #set up interpolator for this ions 
+    xsi2_interp=interpolate.RegularGridInterpolator(mod_axisval,xsi2grid,method='linear',bounds_error=False,fill_value=-np.inf)
+ 
+    #extract x_SiIII
+    xsi3grid=modl[2]['SiIII']
+    #set up interpolator for this ions 
+    xsi3_interp=interpolate.RegularGridInterpolator(mod_axisval,xsi3grid,method='linear',bounds_error=False,fill_value=-np.inf)
+
+    #extract x_CII
+    xc2grid=modl[2]['CII']
+    #set up interpolator for this ions 
+    xc2_interp=interpolate.RegularGridInterpolator(mod_axisval,xc2grid,method='linear',bounds_error=False,fill_value=-np.inf)
+    
+    #extract x_CIII
+    xc3grid=modl[2]['CIII']
+    #set up interpolator for this ions 
+    xc3_interp=interpolate.RegularGridInterpolator(mod_axisval,xc3grid,method='linear',bounds_error=False,fill_value=-np.inf)
+
+    #extract x_CIV
+    xc4grid=modl[2]['CIV']
+    #set up interpolator for this ions 
+    xc4_interp=interpolate.RegularGridInterpolator(mod_axisval,xc4grid,method='linear',bounds_error=False,fill_value=-np.inf)
+
+    #return
+    interp={'HI':xhi_interp,'SiII':xsi2_interp,'SiIII':xsi3_interp,'CII':xc2_interp,'CIII':xc3_interp,'CIV':xc4_interp}
+   
+    return interp
+
+
+"""
+
+Main function
+
+"""
+
+if __name__ == "__main__":
```

Comment:
I'm not sure, either.  It was meant for creating save files/plots with ionization information, so it's meant to be called from the command line.  But if you're still questioning, ask the boss  ;)